Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabInstanceUtils.h

```diff
@@ -10,39 +10,61 @@
-             * of startInstance), and a string of startInstance's relative path to targetInstance.
-             */
-            AZStd::pair<const Instance*, AZStd::string> GetRelativePathBetweenInstances(
-                const Instance* startInstance, const Instance* targetInstance);
+            //! Climbs up the instance hierarchy tree from the given start instance untils it hits
+            //! the target instance or the root instance.
+            //! @param startInstance The instance as the starting point in an instance hierarchy tree.
+            //! @param targetInstance The instance to climb up to. If not provided, root instance will be hit.
+            //! @return InstanceClimbUpResult that shows the climb-up info.
+            InstanceClimbUpResult ClimbUpToTargetOrRootInstance(const Instance& startInstance,
+                InstanceOptionalConstReference targetInstance = AZStd::nullopt);
+
+            //! Generates a relative path from a parent instance to its child instance.
+            //! @param parentInstance The parent instance that the path points from.
+            //! @param childInstance The child instance that the path points to.
+            //! @return The relative path string. Or empty string if the parent instance is not a valid parent.
+            AZStd::string GetRelativePathBetweenInstances(const Instance& parentInstance, const Instance& childInstance);
 
+            //! Generates a relative path from a list of climbed instances.
+            //! @param climbedInstances The list of climbed instances from bottom to top.
+            //! @return The relative path string.
+            AZStd::string GetRelativePath(const AZStd::vector<InstanceOptionalConstReference>& climbedInstances);
+
+            //! Checks if the child instance is a descendant of the parent instance (or ancestor vice versa).
+            //! @param childInstance The given child instance.
+            //! @param parentInstance The given parent instance.
+            //! @return bool on whether the relation is valid.
+            bool IsHierarchical(const Instance& childInstance, const Instance& parentInstance);
+
+            //! Checks if the child instance is a proper descendant (not inclusive) of the parent instance (or ancestor vice versa).
+            //! @param childInstance The given child instance.
+            //! @param parentInstance The given parent instance.
+            //! @return bool on whether the relation is valid.
+            bool IsProperlyHierarchical(const Instance& childInstance, const Instance& parentInstance);
+
+            //! Checks if two instances are identical by address.
+            //! @param instanceA The given instance.
+            //! @param instanceB The given instance to be compared with.
+            //! @return bool on whether the two instances are idendical or both are nullopt.
+            bool CompareInstances(InstanceOptionalConstReference instanceA, InstanceOptionalConstReference instanceB);
```

Comment:
Is this really needed? Seems like you are just comparing pointers here. Seems such a basic operation that this abstraction might be a bit of overkill.

Reply:
imo it does good to readability by hiding `->get()` and `&()`. Currently, there are 3 places using this.

Before:
```cpp
if (&(climbUpResult.reachedInstance->get()) != &(focusedInstance->get()))
{
    AZ_Error("Prefab", false, "PrefabFocusHandler::AppendPathFromFocusedInstanceToPatchPaths - "
        "Entity id is not owned by a descendant of the focused prefab instance.");
    return InvalidLinkId;
}
```

After:
```cpp
if (!PrefabInstanceUtils::CompareInstances(climbUpResult.reachedInstance, focusedInstance))
{
    AZ_Error("Prefab", false, "PrefabFocusHandler::AppendPathFromFocusedInstanceToPatchPaths - "
        "Entity id is not owned by a descendant of the focused prefab instance.");
    return InvalidLinkId;
}
```

Reply:
I also think it is kinda overkill here.
If you just want to increase readability, maybe just declare local variables with good names for the pointers and see if they are the same.